src/test/java/nextstep/subway/unit/SectionTest.java

```diff
@@ -0,0 +1,118 @@
+
+import static org.assertj.core.api.Assertions.*;
+
+@DisplayName("지하철 노선 구간 단위테스트")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class SectionTest {
+    private Line 인천1호선;
+    private Station 계양역;
+    private Station 국제업무지구역;
+    private Station 인천터미널역;
+    private Station 송도달빛축제공원역;
+    private Station 신검암중앙역;
+
+    @BeforeEach
+    void setup() {
+        //given
+        계양역 = new Station(1L, "계양역");
+        국제업무지구역 = new Station(2L, "국제업무지구역");
+        Section section = new Section(계양역, 국제업무지구역, 15);
+        인천1호선 = new Line("인천1호선", "bg-blue-400", section);
+
+        인천터미널역 = new Station(3L, "인천터미널역");
+        신검암중앙역 = new Station(4L, "신검암중앙역");
+        송도달빛축제공원역 = new Station(5L, "송도달빛축제공원역");
+    }
+
+
+    @DisplayName("새 구간을 노선의 끝에 추가한다.")
+    @Test
+    void successAddToEnd() {
+        //when
+        Section 신구간 = new Section(국제업무지구역, 송도달빛축제공원역, 3);
+
+        인천1호선.addSection(신구간);
+
+        //then
+        assertThat(인천1호선.getSections()).last().isEqualTo(신구간);
+    }
+
+    @DisplayName("새 구간을 노선의 처음에 추가한다.")
+    @Test
+    void successAddToFirst() {
+        //when
+        Section 신구간 = new Section(신검암중앙역, 계양역, 3);
+
+        인천1호선.addSection(신구간);
+
+        //then
+        assertThat(인천1호선.getSections()).first().isEqualTo(신구간);
+    }
+
+    @DisplayName("상행역이 동일한 새 구간을 노선의 가운데에 추가한다.")
+    @Test
+    void successAddToMiddleUpStaionCase() {
+        //when
+        Section 신구간 = new Section(계양역, 인천터미널역, 3);
+
+        인천1호선.addSection(신구간);
+
+        //then
+        assertThat(인천1호선.getSections()).containsAnyOf(신구간);
+    }
+
+    @DisplayName("하행역이 동일한 새 구간을 노선의 가운데에 추가한다.")
+    @Test
+    void successAddToMiddleDownStationCase() {
+        //when
+        Section 신구간 = new Section(인천터미널역, 국제업무지구역, 3);
+
+        인천1호선.addSection(신구간);
+
+        //then
+        assertThat(인천1호선.getSections()).containsAnyOf(신구간);
+    }
+
+    @DisplayName("노선 가운데에 추가되는 구간의 거리가 노선의 거리보다 짧지 않으면 구간 추가에 실패한다.")
+    @Test
+    void failAddWhenInvalidDistance() {
+        //when
+        Section 신구간 = new Section(계양역, 인천터미널역, 16);
+
+        //then
+        assertThatThrownBy(() -> 인천1호선.addSection(신구간)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+
+    @DisplayName("구간이 1개인 경우 역을 삭제할 수 없다.")
+    @Test
+    void deleteFailure() {
+        //given
+        Station deleteTarget = 국제업무지구역;
+        assertThatThrownBy(() -> 인천1호선.deleteSection(deleteTarget)).isInstanceOf(NoSuchElementException.class);
+    }
+
+    @DisplayName("정상 구간을 제거한다.")
+    @Test
+    void successDelete() {
+        //when
+        Station newUpStation = 국제업무지구역;
+        Station newDownStation = new Station(3L, "송도달빛축제공원역");
+        Section newSection = new Section(newUpStation, newDownStation, 3);
+        인천1호선.addSection(newSection);
+
+        인천1호선.deleteSection(newDownStation);
+        assertThat(인천1호선.getSections().get(0)).isEqualTo(new Section(계양역, 국제업무지구역, 15));
```

Comment:
아래와 같이 `@BeforeEach`로 초기화한 값들을 그대로 활용해볼 수 있지 않을까요?
```suggestion
        //given
        인천1호선.addSection(new Section(국제업무지구역, 송도달빛축제공원역, 3));

        //when
        인천1호선.deleteSection(송도달빛축제공원역);

        //then
        assertThat(인천1호선.getSections().get(0)).isEqualTo(new Section(계양역, 국제업무지구역, 15));
```